File: src/main/java/io/github/svegon/capi/mixininterface/IClientPlayNetworkHandler.java
package io.github.svegon.capi.mixininterface;

import net.minecraft.client.network.ClientDynamicRegistryType;
import net.minecraft.client.network.PlayerListEntry;
import net.minecraft.client.world.ClientWorld;
import net.minecraft.registry.CombinedDynamicRegistries;
import net.minecraft.registry.DynamicRegistryManager;

import java.util.Map;
import java.util.UUID;

public interface IClientPlayNetworkHandler {
    Map<UUID, PlayerListEntry> playerListEntries();

    void setWorld(ClientWorld world);

    ClientWorld.Properties getWorldProperties();

    void setWorldProperties(ClientWorld.Properties properties);

    void setCombinedDynamicRegistries(DynamicRegistryManager.Immutable combinedDynamicRegistries);
}

File: src/main/kotlin/io/github/svegon/mclientapi/mixininterface/IClientPlayNetworkHandler.kt
package io.github.svegon.capi.mixininterface

import java.util.*

interface IClientPlayNetworkHandler {
    fun playerListEntries(): Map<UUID?, PlayerListEntry?>?

    fun setWorld(world: ClientWorld?)

    fun getWorldProperties(): ClientWorld.Properties?

    fun setWorldProperties(properties: ClientWorld.Properties?)

    fun setCombinedDynamicRegistries(combinedDynamicRegistries: DynamicRegistryManager.Immutable?)
}
